pkg/controller/testutil/test_utils.go

```diff
@@ -349,6 +350,36 @@ func (m *FakeNodeHandler) Patch(_ context.Context, name string, pt types.PatchTy
 	return &updatedNode, nil
 }
 
+// Apply applies a NodeApplyConfiguration to a Node in the fake store.
+func (m *FakeNodeHandler) Apply(ctx context.Context, node *v1apply.NodeApplyConfiguration, opts metav1.ApplyOptions) (*v1.Node, error) {
+	patchOpts := opts.ToPatchOptions()
+	data, err := json.Marshal(node)
+	if err != nil {
+		return nil, err
+	}
+	name := node.Name
```

Comment:
I mentioned that in the other PR, but should `Name` be here or in metadata?

Reply:
Also, you end-up dereferencing the pointer below, should we panic to offer a clearer message?

Reply:
Looks like I check for nil for name but not node. I'll fix that.

See other PR for details about name and metadata. I put an explanation there.